cfme/control/explorer/alerts.py

```diff
@@ -0,0 +1,158 @@
+# -*- coding: utf-8 -*-
+"""Page model for Control / Explorer"""
+from utils.pretty import Pretty
+from utils.appliance.implementations.ui import navigator, navigate_to, CFMENavigateStep
+from navmazing import NavigateToAttribute
+
+from widgetastic.widget import Text, Checkbox
+from widgetastic_patternfly import BootstrapSelect, Button, Input
+
+from . import ControlExplorerView
+from utils.appliance import Navigatable
+from utils.update import Updateable
+
+
+class AlertsAllView(ControlExplorerView):
+    title = Text("#explorer_title_text")
+
+    @property
+    def is_displayed(self):
+        return super(AlertsAllView, self).is_displayed and self.title.text == "All Alerts"
+
+
+class AlertFormCommon(ControlExplorerView):
+    description = Input(name="description")
+    active = Checkbox("enabled_cb")
+    based_on = BootstrapSelect("miq_alert_db")
+    evaluate = BootstrapSelect("exp_name"),
+    driving_event = BootstrapSelect("exp_event")
+    notification_frequency = BootstrapSelect("repeat_time")
+    show_on_timeline = Checkbox("send_evm_event_cb")
+
+    cancel_button = Button('Cancel')
+
+
+class NewAlertView(AlertFormCommon):
+    title = Text("#explorer_title_text")
+
+    add_button = Button("Add")
+
+    @property
+    def is_displayed(self):
+        return (
+            super(NewAlertView, self).is_displayed and
+            self.title.text == "Adding a new Alert" and
+            self.alerts.is_opened and
+            self.alerts.tree.currently_selected == ["All Alerts"]
+        )
+
+
+class EditAlertView(AlertFormCommon):
+    title = Text("#explorer_title_text")
+
+    save_button = Button("Save")
+    reset_button = Button("Reset")
+
+    @property
+    def is_displayed(self):
+        return (
+            super(EditAlertView, self).is_displayed and
+            self.title.text == 'Editing Alert "{}"'.format(self.context["object"].description) and
+            self.alerts.is_opened
+        )
+
+
+class AlertDetailsView(ControlExplorerView):
+    title = Text("#explorer_title_text")
+
+    @property
+    def is_displayed(self):
+        return (
+            super(AlertDetailsView, self).is_displayed and
+            self.title.text == 'Alert "{}"'.format(self.context["object"].description) and
+            self.alerts.is_opened and
+            self.alerts.tree.currently_selected == [
+                "All alerts",
+                self.context["object"].description
+            ]
+        )
+
+
+class Alert(Updateable, Navigatable, Pretty):
+
+    def __init__(self, description, **kwargs):
```

Comment:
@psav this class is also a pretty good potential example for my experiment wrt simpler consistent ctors

as far as i can tell it would beautifully map into the example i provide in #3701